### PR TITLE
Proper Imports Syntax

### PR DESCRIPTION
**Pull request by kustosz**
### Pull Request Description
Imports are now qualified by default and uppercase names can only be resolved if specifically imported.

### Checklist

Please include the following checklist in your PR:

- [x] The documentation has been updated if necessary.
- [x] All code conforms to the [Scala](https://github.com/enso-org/enso/blob/main/docs/style-guide/scala.md), [Java](https://github.com/enso-org/enso/blob/main/docs/style-guide/java.md), and [Rust](https://github.com/enso-org/enso/blob/main/docs/style-guide/rust.md) style guides.
- [x] All documentation and configuration conforms to the [markdown](https://github.com/enso-org/enso/blob/main/docs/style-guide/markdown.md) and [YAML](https://github.com/enso-org/enso/blob/main/docs/style-guide/yaml.md) style guides.
- [x] All code has been tested where possible.

### CODE REVIEW

File: engine/runtime/src/main/scala/org/enso/compiler/data/BindingsMap.scala [hunk 1]
```diff
@@ -65,20 +66,26 @@ case class BindingsMap(
...
-        mod.getIr.unsafeGetMetadata(
-          BindingAnalysis,
-          "Wrong pass ordering. Running resolution on an unparsed module"
-        )
+      .flatMap { imp =>
+        if (imp.importDef.allowsAccess(name)) {
+          imp.module.getIr
+            .unsafeGetMetadata(
+              BindingAnalysis,
+              "Wrong pass ordering. Running resolution on an unparsed module"
```
Comment (iamrecursion) [inline engine/runtime/src/main/scala/org/enso/compiler/data/BindingsMap.scala]:
```suggestion
              "Wrong pass ordering. Running resolution on an unparsed module."
```

File: engine/runtime/src/test/resources/Test_Hiding_Error/src/Atom.enso [hunk 2]
```diff
@@ -0,0 +1,3 @@
+type X a
+
+X.method = 20
```
Comment (iamrecursion) [inline engine/runtime/src/test/resources/Test_Hiding_Error/src/Atom.enso]: Missing newline.

File: engine/runtime/src/test/resources/Test_Hiding_Success/src/Atom.enso [hunk 3]
```diff
@@ -0,0 +1,5 @@
+type X a
+
+type Y b
+
+X.method = 20
```
Comment (iamrecursion) [inline engine/runtime/src/test/resources/Test_Hiding_Success/src/Atom.enso]: Missing newline.

File: engine/runtime/src/test/resources/Test_Qualified_Error/src/Atom.enso [hunk 4]
```diff
@@ -0,0 +1,3 @@
+type X a
+
+X.method = 20
```
Comment (iamrecursion) [inline engine/runtime/src/test/resources/Test_Qualified_Error/src/Atom.enso]: Missing newline.

File: engine/runtime/src/test/resources/Test_Rename/src/Atom.enso [hunk 5]
```diff
@@ -0,0 +1,3 @@
+type X a
+
+X.method = 20
```
Comment (iamrecursion) [inline engine/runtime/src/test/resources/Test_Rename/src/Atom.enso]: Missing newline.

File: engine/runtime/src/test/resources/Test_Rename_Error/src/Atom.enso [hunk 6]
```diff
@@ -0,0 +1,3 @@
+type X a
+
+X.method = 20
```
Comment (iamrecursion) [inline engine/runtime/src/test/resources/Test_Rename_Error/src/Atom.enso]: Missing newline.

**Review by iamrecursion — CHANGES_REQUESTED** [review]
Good stuff, but you're missing the purely syntactic level tests in `AstToIrTest`.

**Review by iamrecursion — APPROVED** [review]

File: lib/scala/syntax/definition/src/main/scala/org/enso/syntax/text/ast/meta/Builtin.scala [hunk 7]
```diff
@@ -267,16 +269,57 @@ object Builtin {
     }
 
     val `import` = {
+      val qualNamePat     = Pattern.SepList(Pattern.Cons(), Opr("."))
+      val rename          = Var("as") :: Pattern.Cons()
+      val all: Pattern    = Var("all")
```
Comment (radeusgd) [inline lib/scala/syntax/definition/src/main/scala/org/enso/syntax/text/ast/meta/Builtin.scala:274]: The `all` pattern seems to not be described in the [imports syntax documentation](https://github.com/enso-org/enso/blob/wip/mk/import-syntax/docs/syntax/imports.md).

Comment (kustosz) [reply lib/scala/syntax/definition/src/main/scala/org/enso/syntax/text/ast/meta/Builtin.scala:274]: None of this whole bunch of work is described. I'm going to update the docs as the last PR in this series.

**Pull request merged**